ci-operator/step-registry/baremetalds/e2e/test/baremetalds-e2e-test-commands.sh

```diff
@@ -28,25 +29,67 @@ IP=$(cat "${SHARED_DIR}/server-ip")
 
 SSHOPTS=(-o 'ConnectTimeout=5' -o 'StrictHostKeyChecking=no' -o 'UserKnownHostsFile=/dev/null' -o 'ServerAliveInterval=90' -i "${CLUSTER_PROFILE_DIR}/.packet-kni-ssh-privatekey")
 
+collect_artifacts() {
+    echo "### Fetching results"
+    ssh "${SSHOPTS[@]}" "root@${IP}" tar -czf - /tmp/artifacts | tar -C "${ARTIFACT_DIR}" -xzf -
+}
+trap collect_artifacts EXIT
+
 # Copy test binaries on packet server
 echo "### Copying test binaries"
 scp "${SSHOPTS[@]}" /usr/bin/openshift-tests /usr/bin/kubectl "root@${IP}:/usr/local/bin"
 
 # Tests execution
 set +e
-if [[ -s "${SHARED_DIR}/test-list" ]]; then
-    echo "### Copying test-list file"
-    scp "${SSHOPTS[@]}" "${SHARED_DIR}/test-list" "root@${IP}:/tmp/test-list"
-    echo "### Running tests"
-    ssh "${SSHOPTS[@]}" "root@${IP}" openshift-tests run "openshift/conformance/parallel" --dry-run \| grep -Ff /tmp/test-list \|openshift-tests run -o /tmp/artifacts/e2e.log --junit-dir /tmp/artifacts/junit -f -
+
+# Test upgrade for workflows that requested it
+if [[ "$RUN_UPGRADE_TEST" == true ]]; then
+    echo "### Running Upgrade tests"
+    # In case of a timeout we don't get the trace log of this script
+    # Let's verify that the we started the upgrade process
+    touch "${ARTIFACT_DIR}/upgrade_started"
+    timeout \
+    --kill-after 10m \
+    120m \
+        ssh \
```

Comment:
If not required anymore, could this timeout be removed (there will be also the CI default one for the job)?

Reply:
I would like to keep it just in case other issues will cause the code to get stuck. In this case, it would be good to have the logs.